tests/integration/test_lock_resolver.py

```diff
@@ -111,21 +111,28 @@ def test_subset(
     # type: (...) -> None
 
     urllib3_pex = os.path.join(str(tmpdir), "urllib3.pex")
-    run_pex_command(
-        args=[
-            "--lock",
-            requests_lock_strict,
-            "urllib3",
-            "-o",
-            urllib3_pex,
-            "--",
-            "-c",
-            "import urllib3",
-        ]
-    ).assert_success()
+    args = [
+        "--lock",
+        requests_lock_strict,
+        "urllib3",
+        "-o",
+        urllib3_pex,
+        "--",
+        "-c",
+        "import urllib3",
+    ]
+    run_pex_command(args).assert_success()
     pex_distributions = index_pex_distributions(urllib3_pex)
-    assert ProjectName("requests") not in pex_distributions
     assert ProjectName("urllib3") in pex_distributions
+    for project in ("requests", "idna", "chardet", "certifi"):
+        assert ProjectName(project) not in pex_distributions
+
+    # However, if no requirements are specified, resolve the entire lock.
+    args.pop(2)
```

Comment:
That's pretty obscure! I'd personally extract a nested function that took optional *requirements. Clearly I made sense of this code as it was though, so there you go.